Link protoc compiler to the libraries.

--- a/vsprojects/protoc.vcproj
+++ b/vsprojects/protoc.vcproj
@@ -65,6 +65,7 @@
 				GenerateDebugInformation="true"
 				SubSystem="1"
 				TargetMachine="1"
+				AdditionalDependencies="$(Configuration)/libprotoc.lib $(Configuration)/libprotobuf.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -137,6 +138,7 @@
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
 				TargetMachine="1"
+				AdditionalDependencies="$(Configuration)/libprotoc.lib $(Configuration)/libprotobuf.lib"
 			/>
 			<Tool
 				Name="VCALinkTool"
